### PR TITLE
feat: create new post route init

### PR DESCRIPTION
**Pull request by YoshkiMatryoshki**
Initial PR for creating post. We map and insert only base part of post ignoring tags. Tags creation logic and transactions planned in next PR(s)

Also small changes in contracts with ids

### CODE REVIEW

File: blog-server-api/src/router.rs [hunk 1]
```diff
@@ -77,6 +77,11 @@ pub fn make_router<Extensions: ExtensionsProviderType>(
                         .and_path("/posts")
                         .and_handler(posts::http_handler),
                 )
+                .route(
+                    route::first::Route::with_method(&hyper::Method::POST)
+                        .and_path("/post/new")
```
Comment (tikitko) [inline blog-server-api/src/router.rs:82]: "/post"

File: blog-server-api/src/endpoints/create_post/request_content.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use screw_api::request::{ApiRequestContent, ApiRequestOriginContent};
+use screw_components::dyn_result::DResult;
+use serde::Deserialize;
+use std::sync::Arc;
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreatePostRequestData {
+    pub post_id: i64,
+    pub post_content: String,
```
Comment (tikitko) [inline blog-server-api/src/endpoints/create_post/request_content.rs:12]: 🤡

File: blog-server-api/src/endpoints/create_post/handler.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    post_service: Arc<Box<dyn PostService>>,
+) -> Result<CreatePostContentSuccess, CreatePostContentFailure> {
+    let CreatePostRequestData {
+        post_id,
+        post_content,
+    } = new_post_data.map_err(|e| ValidationError {
+        reason: e.to_string(),
+    })?;
+
+    let test = post_service
```
Comment (tikitko) [inline blog-server-api/src/endpoints/create_post/handler.rs:23]: xyest

**Review by tikitko — CHANGES_REQUESTED** [review]

File: blog-server-api/src/endpoints/create_post/handler.rs (hunk 3 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/endpoints/create_post/handler.rs:23]: Inappropriate communication is unprofessional and goes against our corporate standards. Let's maintain a respectful and professional environment for everyone's benefit. #ProfessionalismMatters

File: blog-server-api/src/router.rs (hunk 1 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/router.rs]: Changed

File: blog-server-api/src/endpoints/create_post/handler.rs (hunk 3 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/endpoints/create_post/handler.rs]: was needed for testing purposes, now removed

File: blog-server-api/src/endpoints/create_post/request_content.rs (hunk 2 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/endpoints/create_post/request_content.rs]: )000

File: blog-server-api/src/endpoints/create_post/handler.rs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        .await
+        .map_err(|e| DatabaseError {
+            reason: e.to_string(),
+        })?
+        .ok_or(InsertFailed)?;
+
+    Ok(created_post.into())
+}
+
+pub async fn http_handler<Extensions>(
```
Comment (tikitko) [inline blog-server-api/src/endpoints/create_post/handler.rs:41]: 🤡 use example from other handlers....

File: blog-server-api/src/endpoints/create_post/response_content_failure.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    }
+}
+
+impl ApiResponseContentFailure for CreatePostContentFailure {
+    fn identifier(&self) -> &'static str {
+        match self {
+            CreatePostContentFailure::DatabaseError { reason: _ } => "POST_DATABASE_ERROR",
+            CreatePostContentFailure::ValidationError { reason: _ } => "POST_VALIDATION_ERROR",
+            CreatePostContentFailure::AlreadyExists => "POST_ALREASY_EXISTS",
+            CreatePostContentFailure::InsertFailed => "COULD_NOT_FIND_CREATED_POST",
```
Comment (tikitko) [inline blog-server-api/src/endpoints/create_post/response_content_failure.rs:30]: 🤡 prefix always related to endpoint name

File: blog-server-services/src/impls/rbatis_post_service.rs [hunk 6]
```diff
@@ -227,11 +243,7 @@ impl PostService for RbatisPostService {
     }
 
     async fn create_post(&self, post: &BasePost) -> DResult<i64> {
-        let insert_result = BasePost::insert(&mut self.rb.clone(), post).await?;
-        let last_insert_id = insert_result
-            .last_insert_id
-            .as_i64()
-            .ok_or::<DError>("wrond last_insert_id".into())?;
-        Ok(last_insert_id)
+        let test = RbatisPostService::insert_new_post(&self.rb, post).await?;
```
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:246]: 🤡 test?

File: blog-server-services/src/traits/post_service.rs [hunk 7]
```diff
@@ -5,6 +5,7 @@ use serde::{Deserialize, Serialize};
 #[serde(rename_all = "snake_case")]
 pub struct Tag {
     pub id: i64,
```
Comment (tikitko) [inline blog-server-services/src/traits/post_service.rs:7]: u64 🤡

File: blog-server-services/src/impls/rbatis_post_service.rs [hunk 8]
```diff
@@ -15,6 +15,7 @@ impl_insert!(BasePost {}, "post");
 pub struct TagDto {
     post_id: i64,
```
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:16]: u64 🤡

File: blog-server-services/src/impls/rbatis_post_service.rs [hunk 9]
```diff
@@ -15,6 +15,7 @@ impl_insert!(BasePost {}, "post");
...
     id: i64,
```
Comment (tikitko) [inline blog-server-services/src/impls/rbatis_post_service.rs:17]: u64 🤡

File: blog-server-api/src/entities/tag.rs [hunk 10]
```diff
@@ -1,8 +1,9 @@
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Tag {
+    pub id: i64,
```
Comment (tikitko) [inline blog-server-api/src/entities/tag.rs:6]: u64 🤡

File: blog-server-api/src/endpoints/create_post/handler.rs [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+async fn handler(
+    new_post_data: DResult<CreatePost>,
+    post_service: Arc<Box<dyn PostService>>,
+) -> Result<CreatePostContentSuccess, CreatePostContentFailure> {
+    let base_post = new_post_data
+        .map_err(|e| ValidationError {
+            reason: e.to_string(),
+        })?
+        .into(1);
```
Comment (tikitko) [inline blog-server-api/src/endpoints/create_post/handler.rs:21]: [QUESTION] 🤡 WAT?

**Review by tikitko — CHANGES_REQUESTED** [review]

File: blog-server-api/src/endpoints/create_post/handler.rs (hunk 4 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/endpoints/create_post/handler.rs]: done

File: blog-server-api/src/endpoints/create_post/response_content_failure.rs (hunk 5 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/endpoints/create_post/response_content_failure.rs]: Changed

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 6 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: changed

File: blog-server-services/src/traits/post_service.rs (hunk 7 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/traits/post_service.rs]: changed, also for some methods and other entities

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 8 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: changed, also for some methods and other entities

File: blog-server-services/src/impls/rbatis_post_service.rs (hunk 9 above)
Comment (YoshkiMatryoshki) [reply blog-server-services/src/impls/rbatis_post_service.rs]: changed, also for some methods and other entities

File: blog-server-api/src/entities/tag.rs (hunk 10 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/entities/tag.rs]: changed, also for some methods and other entities

File: blog-server-api/src/endpoints/create_post/handler.rs (hunk 11 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/endpoints/create_post/handler.rs]: just simplification for db related processes testing. Now i added author id for request body which is also simplification for now. (TODO placed). 

If you pass request with unexisting author we go straight to the database error 🤡🤡🤡🤡🤡

File: blog-server-api/src/entities/create_post.rs [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreatePost {
+    pub title: String,
+    pub slug: String,
+    pub summary: String,
+    pub published: u8,
+    pub content: Option<String>,
+    pub tags: Vec<ShortTag>,
+    pub author_id: u64, //TODO replace with authorized user ID
```
Comment (tikitko) [inline blog-server-api/src/entities/create_post.rs:15]: We already can do this

Comment (tikitko) [reply blog-server-api/src/entities/create_post.rs:15]: Why not?)

**Review by tikitko — DISMISSED** [review]

File: blog-server-api/src/entities/create_post.rs (hunk 12 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/entities/create_post.rs]: was facing some complications (already discussed). now it is done

**Review by tikitko — APPROVED** [review]

**Review by tikitko — APPROVED** [review]

**Pull request merged**